server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -573,60 +573,69 @@ private boolean poolOverlapsRange(Pool p, Date startDate, Date endDate) {
+                "     SET dirty = true" +
+                "     WHERE id IN (" +
+                "         SELECT * FROM (" +
+                "             SELECT distinct ent.id" +
+                "             FROM cp_entitlement ent" +
+                "             INNER JOIN cp_pool pool" +
+                "                 ON ent.pool_id = pool.id" +
+                "             INNER JOIN cp2_pool_provided_products providedPr" +
+                "                 ON pool.id = providedPr.pool_id" +
+                "             INNER JOIN cp2_products product" +
+                "                 ON providedPr.product_uuid = product.uuid" +
+                "             INNER JOIN cp2_product_content productCon" +
+                "                 ON product.uuid = productCon.product_uuid" +
+                "             INNER JOIN cp2_content content" +
+                "                 ON productCon.content_uuid = content.uuid" +
+                "             INNER JOIN cp2_content_modified_products modifiedPr" +
+                "                 ON content.uuid=modifiedPr.content_uuid" +
+                "             WHERE pool.endDate >= current_date" +
+                "               AND ent.owner_id = :owner" +
+                "               AND ent.dirty = false" +
+                "               AND (ent.id NOT IN (:ein))" +
+                "               AND (EXISTS (SELECT entIn.id" +
+                "                            FROM cp_entitlement entIn" +
+                "                            INNER JOIN cp_consumer consumerIn" +
+                "                                ON entIn.consumer_id = consumerIn.id" +
+                "                            INNER JOIN cp_pool poolIn" +
+                "                                ON entIn.pool_id = poolIn.id" +
+                "                            INNER JOIN cp2_products productIn" +
+                "                                ON poolIn.product_uuid = productIn.uuid" +
+                "                            LEFT OUTER JOIN cp2_pool_provided_products providedPrIn" +
+                "                                ON poolIn.id = providedPrIn.pool_id" +
+                "                            LEFT OUTER JOIN cp2_products productIn2" +
+                "                                ON providedPrIn.product_uuid = productIn2.uuid" +
+                "                            WHERE (entIn.id IN (:ein))" +
+                "                               AND consumerIn.id = ent.consumer_id" +
+                "                               AND poolIn.endDate >= pool.startDate" +
+                "                               AND poolIn.startDate <= pool.endDate" +
+                "                               AND (productIn2.product_id = modifiedPr.element " +
+                "                                 OR productIn.product_id = modifiedPr.element)))) as e)";
```

Comment:
@Ceiu  seems you got it right, just one difference:
"and the entitlement ID is **NOT** in the given set of entitlements"

if it helps, I used the hibernate generated sql queries from the logs to write this query and then tweaked it to make it work, so perhaps doing the same might make the review easier?